### PR TITLE
Server, user and bot info commands

### LINKED ISSUE #81: User info command

### LINKED ISSUE #82: Server info command

### LINKED ISSUE #83: Bot info command

### PR DESCRIPTION
**Pull request by bolsomito**
+ Closes #81 
+ Closes #82 
+ Closes #83 

s!botinfo
s!serverinfo
s!userinfo

*emotes can be found here: https://discord.gg/czXM8rp*

### CODE REVIEW

File: src/commands/misc/botinfo.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  run ({ channel, author, t }) {
+    const embed = new SwitchbladeEmbed(author)
+    const uptime = moment.duration(process.uptime() * 1000).format('d[d] h[h] m[m] s[s]')
+    channel.startTyping()
+
+    embed.setAuthor(this.client.user.username, this.client.user.displayAvatarURL)
+      .setThumbnail(this.client.user.displayAvatarURL)
+      .setDescription([
+        t('commands:botinfo.hello', { this: this }),
```
Comment (davipatury) [inline src/commands/misc/botinfo.js]: I think we should just send the information that is used in the string. People could access undesired content if we send the whole command.

File: src/commands/misc/botinfo.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        t('commands:botinfo.statistics', { this: this, uptime, Discord, process })
```
Comment (davipatury) [inline src/commands/misc/botinfo.js]: Same as above, but here the problem is the `process` variable.

Comment (bolsomito) [reply src/commands/misc/botinfo.js]: keep on reviewing

File: src/commands/misc/userinfo.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    this.parameters = new CommandParameters(this,
+      new MemberParameter({full: true, required: false, acceptBot: true})
+    )
+  }
+
+  run ({ t, member: author, channel, guildDocument }, member) {
+    const embed = new SwitchbladeEmbed(author)
+    member = member || author
+    moment.locale(guildDocument.language)
+    const filter = this.client.guilds.filter(g => g.members.get(member.id)).map(g => g.name)
```
Comment (davipatury) [inline src/commands/misc/userinfo.js]: `const filter = this.client.guilds.filter(g => g.members.has(member.id))`

File: src/commands/misc/userinfo.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    channel.startTyping()
+
+    embed.setTitle(member.displayName)
+      .setThumbnail(member.user.displayAvatarURL)
+      .addField(t('commands:userinfo.tag'), member.user.tag, true)
+      .addField(t('commands:userinfo.id'), member.id, true)
+      .addField(t('commands:userinfo.status'), t(`commands:userinfo.${member.presence.status}`, { Constants }), true)
+      .addField(t('commands:userinfo.createdAt'), moment(member.user.createdTimestamp).format('LLL'), true)
+      .addField(t('commands:userinfo.joinedAt'), moment(member.joinedTimestamp).format('LLL'), true)
+      .addField(t('commands:userinfo.serversInCommon', { count: filter.length }), filter.join(', '))
```
Comment (davipatury) [inline src/commands/misc/userinfo.js]: `{ count: filter.size }`

File: src/locales/en-US/commands.json [hunk 5]
```diff
@@ -225,6 +225,48 @@
...
+    "offline": "Offline: **{{count}}**",
+    "users": "Users: **{{count}}**",
+    "bots": "Bots: **{{count}}**"
+  },
+  "userinfo": {
+    "commandDescription": "Sends your or someone else's user information",
+    "online": "{{Constants.ONLINE_STATUS}} Online",
+    "idle": "{{Constants.IDLE_STATUS}} Idle",
+    "dnd": "{{Constants.DND_STATUS}} Do Not Disturb",
+    "offline": "{{Constants.OFFLINE_STATUS}} Offline",
```
Comment (davipatury) [inline src/locales/en-US/commands.json]: Remove the `{{Contants.XXX}}`, it's not used.

**Comment by bolsomito** [comment]
# Ready for review

**Pull request merged**